### ISSUE TITLE
Provide advice about proxies receiving UDP packets that won't fit in datagrams

### ISSUE THREAD

**Issue by tfpauly** [ietf-wg-masque/draft-ietf-masque-connect-udp#40]
A proxy can receive a UDP packet from an origin to a client that cannot fit within a DATAGRAM frame, given either the limit set by the maximum datagram frame size, or based on the current path MTU.

In this case, one solution is for the proxy to generate ICMP errors back to the sender. If that sender implements PMTU discovery, it should adjust its size.

**Comment by DavidSchinazi** [comment]
I think adding advice here is good, though we can't mandate sending ICMP because most OSes don't let user space applications send ICMP.

**Comment by tfpauly** [comment]
Yes, we can't mandate it, but we should recommend it I think. For a full-blown production server, it should be possible.

**Comment by LPardue** [comment]
Recommendation seems sensible. Although this does get me thinking about what the expectations of fragmentation are, we can't expect all UDP protocols to set DF.

**Comment by MikeBishop** [comment]
The DATAGRAM capsule type offers one way to address this.  Capsules can be arbitrarily large, because they'll be reassembled by QUIC.  If it won't fit in a QUIC DATAGRAM and you don't want to drop it, it goes in a DATAGRAM capsule.

Now, that might not always be desirable, because capsules don't have all the behavior that causes you to choose datagrams in the first place; a proxy that behaves that way will mess with PMTUD, and in the degenerate case cause the server to select an MTU back to the client that _always_ gets encoded in capsules.

Perhaps the client should be able to inform the server whether it prefers oversize messages to be tunneled or dropped?
